crates/router/src/core/payments/helpers.rs

```diff
@@ -2037,6 +2037,27 @@ pub fn decide_payment_method_retrieval_action(
     }
 }
 
+pub async fn is_ucs_enabled(state: &SessionState, config_key: &str) -> bool {
+    let db = state.store.as_ref();
+    let is_enabled = db
+        .find_config_by_key_unwrap_or(config_key, Some("false".to_string()))
+        .await
+        .map_err(|err| {
+            logger::error!(error = ?err, "Failed to fetch {config_key:?} UCS enabled config from DB");
+        })
+        .ok()
+        .and_then(|config| {
+            config.config.parse::<bool>()
+                .map_err(|err| {
+                    logger::error!(error = ?err, "Failed to parse {config_key:?} UCS enabled config");
+                })
+                .ok()
+        })
+        .unwrap_or(false);
+
+    is_enabled
+}
```

Comment:
Nits:

1. You can remove the `ok()` calls.
2. You can use the `Display` impl of `config_key`, since it's a `&str`.
3. If you want to, you can log the value of `config.config` when we fail to parse it (as `bool`), so that we know the value that was present at that point of time.

```suggestion
pub async fn is_ucs_enabled(state: &SessionState, config_key: &str) -> bool {
    let db = state.store.as_ref();
    let is_enabled = db
        .find_config_by_key_unwrap_or(config_key, Some("false".to_string()))
        .await
        .map_err(|error| {
            logger::error!(?error, "Failed to fetch `{config_key}` UCS enabled config from DB");
        })
        .and_then(|config| {
            config.config.parse::<bool>()
                .map_err(|error| {
                    logger::error!(?error, "Failed to parse `{config_key}` UCS enabled config");
                })
        })
        .unwrap_or(false);

    is_enabled
}
```